### PR TITLE
gh-126899: Add `**kw` to `tkinter.Misc.after` and `tkinter.Misc.after_idle`

### PR DESCRIPTION
**Pull request by Xiaokang2022**
* Issue: gh-126899

### CODE REVIEW

**Comment by graingert** [comment]
For this to work you need to make func and ms positional only, which is backwards incompatible

**Comment by Xiaokang2022** [comment]
> For this to work you need to make func and ms positional only, which is backwards incompatible

Is making `func` and `ms` positional only a must? If not, it will not lead to incompatibilities.

I've found something similar elsewhere in `tkinter`, but it doesn't enforce that the preceding argument to positional only:

https://github.com/python/cpython/blob/2313f8421080ceb3343c6f5d291279adea85e073/Lib/tkinter/ttk.py#L444

and

https://github.com/python/cpython/blob/2313f8421080ceb3343c6f5d291279adea85e073/Lib/tkinter/ttk.py#L547

**Comment by serhiy-storchaka** [comment]
It is not necessary to make other parameters positional only, but this can be considered in future, together with other methods that have var-keyword parameter.

In general, I do not see objections against this change. It does not also look that other Tkinter methods need similar change. But please add tests and a What's New entry.

File: Doc/whatsnew/3.14.rst [hunk 1]
```diff
@@ -514,6 +514,15 @@ sys
   from other interpreters than the one it's called in.
 
 
+tkinter
+-------
+
+* Add the argument ``**kw`` to method :meth:`!after` and method
+  :meth:`!after_idle` of :mod:`tkinter` widgets so that the keyword argument
+  can passed to ``func`` Conveniently.
```
Comment (Wulian233) [inline Doc/whatsnew/3.14.rst]:
```suggestion
  can passed to ``func`` conveniently.
```

Comment (Xiaokang2022) [reply Doc/whatsnew/3.14.rst]: The keyword argument here does not refer to `func`.

**Comment by Xiaokang2022** [comment]
@serhiy-storchaka PTAL, thanks!

File: Doc/whatsnew/3.14.rst [hunk 2]
```diff
@@ -514,6 +514,15 @@ sys
   from other interpreters than the one it's called in.
 
 
+tkinter
+-------
+
+* Add the argument ``**kw`` to method :meth:`!after` and method
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.14.rst]: It is a var-keyword parameter, not argument (see https://docs.python.org/3/glossary.html#term-parameter).

I would write something like "The ... methods of ... now support passing keyword aguments to the function." In any case this needs a review by a native English speaker.

File: Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Add ``**kw`` to ``tkinter.Misc.after`` and ``tkinter.Misc.after_idle``.
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst]:
```suggestion
Add the var-positional parameter to :meth:`!tkinter.Misc.after` and :meth:`!tkinter.Misc.after_idle`.
```

File: Doc/whatsnew/3.14.rst (hunk 2 above)
Comment (Xiaokang2022) [reply Doc/whatsnew/3.14.rst]: Sorry, my translator isn't particularly accurate.

File: Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst (hunk 3 above)
Comment (Xiaokang2022) [reply Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst]: It shouldn't be `var-positional`, it's `var-keyword`, right?

Comment (serhiy-storchaka) [reply Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst]: Right. My bad. :man_facepalming:

**Comment by Xiaokang2022** [comment]
> In any case this needs a review by a native English speaker.

@terryjreedy , As [mentioned above](https://github.com/python/cpython/pull/126900#discussion_r1846273234), I guess you may be able to help us review this PR, it's easy, thanks!

File: Doc/whatsnew/3.14.rst [hunk 4]
```diff
@@ -514,6 +514,14 @@ sys
   from other interpreters than the one it's called in.
 
 
+tkinter
+-------
+
+* The methods :meth:`!tkinter.Misc.after` and :meth:`!tkinter.Misc.after_idle`
+  now support passing keyword arguments to the function.
```
Comment (terryjreedy) [inline Doc/whatsnew/3.14.rst]: Misc is effectively a mixin class that is never instantiated and whose existence is an implementation detail.  People always call these methods on widgets.

```suggestion
* Make tkinter widget methods ``after`` and ``after_idle`` accept arguments passed by keyword.
```
There are other possible wordings, but as a tkinter user, this seems clear enough to me.

File: Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Add the var-keyword parameter to :meth:`!tkinter.Misc.after` and :meth:`!tkinter.Misc.after_idle`.
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Library/2024-11-16-10-52-48.gh-issue-126899.GFnfBt.rst]: Revise as suggested for what's new.
```suggestion
Make tkinter widget methods ``after`` and ``after_idle`` accept arguments passed by keyword.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I leave code changes to Serhiy.

File: Doc/whatsnew/3.14.rst (hunk 4 above)
Comment (serhiy-storchaka) [reply Doc/whatsnew/3.14.rst]: I would use formatting ``` :meth:`!after` ```.

**Review by serhiy-storchaka — COMMENTED** [review]
The code LGTM.

**Comment by Xiaokang2022** [comment]
I have made the requested changes; please review again.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
There was a (rare) timeout error in building Python for the test.  Nothing to do with us.  Re-running the failed test only.

**Pull request merged**